README.md

```diff
@@ -1,2 +1,174 @@
-# module-ballerinax-shopify
-Ballerina Shopify module
+![Shopify Continuous Integration Workflow](https://github.com/ThisaruGuruge/module-ballerinax-shopify/workflows/Shopify%20Continuous%20Integration%20Workflow/badge.svg)
+# Ballerina Shopify Connector
+This module allows to access Shopify admin REST API though Ballerina. Shopify is a popular E-commerce platform which enables users to create online stores easily. The Shopify admin API provides various functionalities for handle Shopify stores.
+
+The following sections provide you details on how to use the Shopify connector.
+
+- [Compatibility](#compatibility)
+- [Feature Overview](#feature-overview)
+- [Prerequisites](#Prerequisites)
+- [Samples](#samples)
+
+## Compatibility
+
+| Ballerina Shopify Connector Version | Shopify Admin API Version | Ballerina Version |
+|:-----------------------------------:|:-------------------------:|:-----------------:|
+| 0.1.0                               | 2020-04                   | 1.2.x             |
+
+## Feature Overview
+- Shopify connector has three different clients to handle different operations.
+    - `shopify:CustomerClient`
+    - `shopify:ProductClient`
+    - `shopify:OrderClient`
+- The `shopify:Store` object is the base of all the above clients.
+- Basic (Username and Password) and OAuth (Access Token) authentication are supported.
+
+## Prerequisites
+- Download and install the compatible [Ballerina](https://ballerinalang.org/downloads/) distribution.
+- Pull the Shopify module from the Ballerina central
+```shell
+ballerina pull ballerinax/shopify
+```
+- You must obtain the authentication from [Shopify Admin API](https://shopify.dev/concepts/about-apis/authentication).
+
+## Samples
+### Shopify Customer Client
+The `shopify:CustomerClient` client object can be used to handle various Customer-related operations including,
+- Create Customers
+- Retrieve Customer Details
+- Update Customer Details
+- Delete Customers
+
+```ballerina
+import ballerina/io;
+import ballerinax/shopify;
+
+public function main() {
+    io:println("Shopify Customer");
+    string token = <Your Shopify Access Token>;
+    shopify:OAuthConfiguration oAuthConfiguration = {
+        accessToken: token
+    };
+    shopify:StoreConfiguration storeConfiguration = {
+        storeName: <Your Shopify Store Name>,
+        authConfiguration: oAuthConfiguration
+    };
+    // Create a Store object
+    shopify:Store store = new (storeConfiguration);
+    // Obtain the CustomerClient from the Store object
+    shopify:CustomerClient customerClient = store.getCustomerClient();
+
+    shopify:CustomerFilter filter = {
+        // Retrieve only specific fields of a Customer
+        fields: ["firstName", "lastName", "email", "id"]
+    };
+    // Get all the customers of the shop as a stream.
```

Comment:
```suggestion
    // Get all the customers of the shop as a stream
```
If you are using the fullstop at the end of the comment, be consistent and use it in all the others as well.